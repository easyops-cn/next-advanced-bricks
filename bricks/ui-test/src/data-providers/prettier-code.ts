import { createProviderClass } from "@next-core/utils/general";
import { Options } from "prettier";
import { formatCode } from "../utils/parseSuiteAst";

export async function prettierCode(
  source: string,
  options?: Options
): Promise<unknown> {
  return formatCode(source, options);
}

customElements.define(
  "ui-test.prettier-code",
  createProviderClass(prettierCode)
);
