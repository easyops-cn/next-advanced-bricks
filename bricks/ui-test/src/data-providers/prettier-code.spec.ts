import { describe, test, expect } from "@jest/globals";
import { prettierCode } from "./prettier-code.js";

describe("prettierCode", () => {
  test("should work", async () => {
    const code =
      'describe("new", () => {beforeEach(() => {cy.setLanguage("en");cy.login();});it("test", () => {cy.get("test").should("be.empty").contains("test"); cy.get("test");});});';
    expect(await prettierCode(code)).toMatchSnapshot();
  });
});
