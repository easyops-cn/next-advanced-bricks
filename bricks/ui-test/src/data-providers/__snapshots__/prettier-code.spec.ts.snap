// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prettierCode should work 1`] = `
"describe("new", () => {
  beforeEach(() => {
    cy.setLanguage("en");
    cy.login();
  });
  it("test", () => {
    cy.get("test")
      .should("be.empty")
      .contains("test");
    cy.get("test");
  });
});
"
`;
